compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -3872,22 +3872,50 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                 param.name.ident(),
                             ));
                             let bounds_span = hir_generics.bounds_span_for_suggestions(def_id);
+                            let mut applicability = Applicability::MaybeIncorrect;
+                            // Format the path of each suggested candidate, providing placeholders
+                            // for any generic arguments without defaults.
+                            let candidate_strs: Vec<String> = candidates
+                                .iter()
+                                .map(|cand| {
+                                    use ty::GenericParamDefKind::{Const, Lifetime, Type};
```

Comment:
You should be able to simplify this by using `identity_for_item` over `generics_of`. Compare:

https://github.com/rust-lang/rust/blob/ef972a346668ed4234d1a43ed4ad7ca4e9c58d51/compiler/rustc_hir_analysis/src/hir_ty_lowering/errors.rs#L253-L263

Reply:
Ah, disregard, that wouldn't consider generic parameter defaults.